src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -694,5 +693,16 @@ private CorInfoHelpFunc getNewArrHelper(CORINFO_CLASS_STRUCT_* arrayCls)
         {
             return CorInfoHelpFunc.CORINFO_HELP_NEWARR_1_DIRECT;
         }
+
+        private bool canTailCall(CORINFO_METHOD_STRUCT_* callerHnd, CORINFO_METHOD_STRUCT_* declaredCalleeHnd, CORINFO_METHOD_STRUCT_* exactCalleeHnd, bool fIsTailPrefix)
+        {
+            if (fIsTailPrefix)
+            {
+                throw new RequiresRuntimeJitException(nameof(fIsTailPrefix));
+            }
+
+            // No restrictions on tailcalls
```

Comment:
FWIW, crossgen R2R returns `false` here today. Is it intentional that this returns `true`?

Reply:
Of course you're right, I cannot read :-). I'm going to close this PR as its primary goal has been completely nullified by your CoreCLR change to support RVA fields and I'll send out the canTailCall / getFunctionEntryPoint change in a separate PR.